diff --git a/node_modules/@tanstack/query-devtools/build/chunk/6EPU3DES.js b/node_modules/@tanstack/query-devtools/build/chunk/6EPU3DES.js
index 66c53e2..7055de4 100644
--- a/node_modules/@tanstack/query-devtools/build/chunk/6EPU3DES.js
+++ b/node_modules/@tanstack/query-devtools/build/chunk/6EPU3DES.js
@@ -2830,7 +2830,7 @@ function getQueryStatusColor({
   observerCount,
   isStale
 }) {
-  return queryState.fetchStatus === "fetching" ? "blue" : !observerCount ? "gray" : queryState.fetchStatus === "paused" ? "purple" : isStale ? "yellow" : "green";
+  return queryState?.fetchStatus === "fetching" ? "blue" : !observerCount ? "gray" : queryState?.fetchStatus === "paused" ? "purple" : isStale ? "yellow" : "green";
 }
 function getMutationStatusColor({
   status,
diff --git a/node_modules/@tanstack/query-devtools/build/chunk/LNJC7UVE.js b/node_modules/@tanstack/query-devtools/build/chunk/LNJC7UVE.js
index d99a14a..59b1755 100644
--- a/node_modules/@tanstack/query-devtools/build/chunk/LNJC7UVE.js
+++ b/node_modules/@tanstack/query-devtools/build/chunk/LNJC7UVE.js
@@ -2829,7 +2829,7 @@ function getQueryStatusColor({
   observerCount,
   isStale
 }) {
-  return queryState.fetchStatus === "fetching" ? "blue" : !observerCount ? "gray" : queryState.fetchStatus === "paused" ? "purple" : isStale ? "yellow" : "green";
+  return queryState?.fetchStatus === "fetching" ? "blue" : !observerCount ? "gray" : queryState?.fetchStatus === "paused" ? "purple" : isStale ? "yellow" : "green";
 }
 function getMutationStatusColor({
   status,
diff --git a/node_modules/@tanstack/query-devtools/build/chunk/WIJZKUEN.jsx b/node_modules/@tanstack/query-devtools/build/chunk/WIJZKUEN.jsx
index 5c1eae6..763cf53 100644
--- a/node_modules/@tanstack/query-devtools/build/chunk/WIJZKUEN.jsx
+++ b/node_modules/@tanstack/query-devtools/build/chunk/WIJZKUEN.jsx
@@ -2842,7 +2842,7 @@ function getQueryStatusColor({
   observerCount,
   isStale
 }) {
-  return queryState.fetchStatus === "fetching" ? "blue" : !observerCount ? "gray" : queryState.fetchStatus === "paused" ? "purple" : isStale ? "yellow" : "green";
+  return queryState?.fetchStatus === "fetching" ? "blue" : !observerCount ? "gray" : queryState?.fetchStatus === "paused" ? "purple" : isStale ? "yellow" : "green";
 }
 function getMutationStatusColor({
   status,
diff --git a/node_modules/@tanstack/query-devtools/build/chunk/YDKHGUIL.jsx b/node_modules/@tanstack/query-devtools/build/chunk/YDKHGUIL.jsx
index c7e3e57..1ba1176 100644
--- a/node_modules/@tanstack/query-devtools/build/chunk/YDKHGUIL.jsx
+++ b/node_modules/@tanstack/query-devtools/build/chunk/YDKHGUIL.jsx
@@ -2843,7 +2843,7 @@ function getQueryStatusColor({
   observerCount,
   isStale
 }) {
-  return queryState.fetchStatus === "fetching" ? "blue" : !observerCount ? "gray" : queryState.fetchStatus === "paused" ? "purple" : isStale ? "yellow" : "green";
+  return queryState?.fetchStatus === "fetching" ? "blue" : !observerCount ? "gray" : queryState?.fetchStatus === "paused" ? "purple" : isStale ? "yellow" : "green";
 }
 function getMutationStatusColor({
   status,
diff --git a/node_modules/@tanstack/query-devtools/src/utils.tsx b/node_modules/@tanstack/query-devtools/src/utils.tsx
index 6159fcd..1ba2319 100644
--- a/node_modules/@tanstack/query-devtools/src/utils.tsx
+++ b/node_modules/@tanstack/query-devtools/src/utils.tsx
@@ -35,11 +35,11 @@ export function getQueryStatusColor({
   observerCount: number
   isStale: boolean
 }) {
-  return queryState.fetchStatus === 'fetching'
+  return queryState?.fetchStatus === 'fetching'
     ? 'blue'
     : !observerCount
       ? 'gray'
-      : queryState.fetchStatus === 'paused'
+      : queryState?.fetchStatus === 'paused'
         ? 'purple'
         : isStale
           ? 'yellow'
