diff --git b/packages/utils/protobuf/codegen/cosmos/authz/v1beta1/tx.ts a/packages/utils/protobuf/codegen/cosmos/authz/v1beta1/tx.ts
index c04740c22..98853e53d 100644
--- b/packages/utils/protobuf/codegen/cosmos/authz/v1beta1/tx.ts
+++ a/packages/utils/protobuf/codegen/cosmos/authz/v1beta1/tx.ts
@@ -3,6 +3,7 @@ import { Grant, GrantAmino, GrantSDKType } from "./authz";
 import { Any, AnyProtoMsg, AnyAmino, AnySDKType } from "../../../google/protobuf/any";
 import { BinaryReader, BinaryWriter } from "../../../binary";
 import { bytesFromBase64, base64FromBytes } from "../../../helpers";
+import { aminoToRawProtobufMsg, rawProtobufMsgToAmino } from '../../../../../messages/protobuf'
 /**
  * MsgGrant is a request type for Grant method. It declares authorization to the grantee
  * on behalf of the granter with the provided expiration time.
@@ -371,14 +372,14 @@ export const MsgExec = {
     if (object.grantee !== undefined && object.grantee !== null) {
       message.grantee = object.grantee;
     }
-    message.msgs = object.msgs?.map(e => Cosmos_basev1beta1Msg_FromAmino(e)) || [];
+    message.msgs = object.msgs?.map(e => aminoToRawProtobufMsg(e)) || [];
     return message;
   },
   toAmino(message: MsgExec, useInterfaces: boolean = false): MsgExecAmino {
     const obj: any = {};
     obj.grantee = message.grantee;
     if (message.msgs) {
-      obj.msgs = message.msgs.map(e => e ? Cosmos_basev1beta1Msg_ToAmino((e as Any), useInterfaces) : undefined);
+      obj.msgs = message.msgs.map(e => e ? rawProtobufMsgToAmino((e as Any), useInterfaces) : undefined);
     } else {
       obj.msgs = [];
     }
diff --git b/packages/utils/protobuf/codegen/cosmos/bank/v1beta1/bank.ts a/packages/utils/protobuf/codegen/cosmos/bank/v1beta1/bank.ts
index 9eb780900..70b01f2fc 100644
--- b/packages/utils/protobuf/codegen/cosmos/bank/v1beta1/bank.ts
+++ a/packages/utils/protobuf/codegen/cosmos/bank/v1beta1/bank.ts
@@ -930,7 +930,9 @@ export const Metadata = {
   },
   toAmino(message: Metadata, useInterfaces: boolean = false): MetadataAmino {
     const obj: any = {};
-    obj.description = message.description;
+    if (message.description) {
+      obj.description = message.description;
+    }
     if (message.denomUnits) {
       obj.denom_units = message.denomUnits.map(e => e ? DenomUnit.toAmino(e, useInterfaces) : undefined);
     } else {
@@ -940,8 +942,12 @@ export const Metadata = {
     obj.display = message.display;
     obj.name = message.name;
     obj.symbol = message.symbol;
-    obj.uri = message.uri;
-    obj.uri_hash = message.uriHash;
+    if (message.uri) {
+      obj.uri = message.uri;
+    }
+    if (message.uriHash) {
+      obj.uri_hash = message.uriHash;
+    }
     return obj;
   },
   fromAminoMsg(object: MetadataAminoMsg): Metadata {
diff --git b/packages/utils/protobuf/codegen/cosmos/gov/v1/tx.ts a/packages/utils/protobuf/codegen/cosmos/gov/v1/tx.ts
index 6c5cc594d..99099f81a 100644
--- b/packages/utils/protobuf/codegen/cosmos/gov/v1/tx.ts
+++ a/packages/utils/protobuf/codegen/cosmos/gov/v1/tx.ts
@@ -1,7 +1,7 @@
 //@ts-nocheck
 import { Any, AnyProtoMsg, AnyAmino, AnySDKType } from "../../../google/protobuf/any";
 import { Coin, CoinAmino, CoinSDKType } from "../../base/v1beta1/coin";
-import { VoteOption, WeightedVoteOption, WeightedVoteOptionAmino, WeightedVoteOptionSDKType, Params, ParamsAmino, ParamsSDKType, voteOptionFromJSON } from "./gov";
+import { VoteOption, WeightedVoteOption, WeightedVoteOptionAmino, WeightedVoteOptionSDKType, Params, ParamsAmino, ParamsSDKType } from "./gov";
 import { Timestamp } from "../../../google/protobuf/timestamp";
 import { MsgCreateAllianceProposal, MsgCreateAllianceProposalProtoMsg, MsgCreateAllianceProposalSDKType, MsgUpdateAllianceProposal, MsgUpdateAllianceProposalProtoMsg, MsgUpdateAllianceProposalSDKType, MsgDeleteAllianceProposal, MsgDeleteAllianceProposalProtoMsg, MsgDeleteAllianceProposalSDKType } from "../../../alliance/alliance/gov";
 import { CommunityPoolSpendProposal, CommunityPoolSpendProposalProtoMsg, CommunityPoolSpendProposalSDKType, CommunityPoolSpendProposalWithDeposit, CommunityPoolSpendProposalWithDepositProtoMsg, CommunityPoolSpendProposalWithDepositSDKType } from "../../distribution/v1beta1/distribution";
@@ -22,6 +22,7 @@ import { AllowDenomProposal, AllowDenomProposalProtoMsg, AllowDenomProposalSDKTy
 import { CreditTypeProposal, CreditTypeProposalProtoMsg, CreditTypeProposalSDKType } from "../../../regen/ecocredit/v1/types";
 import { BinaryReader, BinaryWriter } from "../../../binary";
 import { toTimestamp, fromTimestamp } from "../../../helpers";
+import { aminoToRawProtobufMsg, rawProtobufMsgToAmino } from "../../../../../messages";
 /**
  * MsgSubmitProposal defines an sdk.Msg type that supports submitting arbitrary
  * proposal Content.
@@ -577,7 +578,7 @@ export const MsgSubmitProposal = {
   },
   fromAmino(object: MsgSubmitProposalAmino): MsgSubmitProposal {
     const message = createBaseMsgSubmitProposal();
-    message.messages = object.messages?.map(e => Any.fromAmino(e)) || [];
+    message.messages = object.messages?.map(e => aminoToRawProtobufMsg(e)) || [];
     message.initialDeposit = object.initial_deposit?.map(e => Coin.fromAmino(e)) || [];
     if (object.proposer !== undefined && object.proposer !== null) {
       message.proposer = object.proposer;
@@ -598,10 +599,8 @@ export const MsgSubmitProposal = {
   },
   toAmino(message: MsgSubmitProposal, useInterfaces: boolean = false): MsgSubmitProposalAmino {
     const obj: any = {};
-    if (message.messages) {
-      obj.messages = message.messages.map(e => e ? Any.toAmino(e, useInterfaces) : undefined);
-    } else {
-      obj.messages = [];
+    if (message.messages.length) {
+      obj.messages = message.messages.map(e => e ? rawProtobufMsgToAmino(e, false) : undefined);
     }
     if (message.initialDeposit) {
       obj.initial_deposit = message.initialDeposit.map(e => e ? Coin.toAmino(e, useInterfaces) : undefined);
@@ -612,7 +611,9 @@ export const MsgSubmitProposal = {
     obj.metadata = message.metadata;
     obj.title = message.title;
     obj.summary = message.summary;
-    obj.expedited = message.expedited;
+    if (message.expedited) {
+      obj.expedited = message.expedited;
+    }
     return obj;
   },
   fromAminoMsg(object: MsgSubmitProposalAminoMsg): MsgSubmitProposal {
@@ -911,7 +912,7 @@ export const MsgVote = {
       message.voter = object.voter;
     }
     if (object.option !== undefined && object.option !== null) {
-      message.option = voteOptionFromJSON(object.option);
+      message.option = object.option;
     }
     if (object.metadata !== undefined && object.metadata !== null) {
       message.metadata = object.metadata;
diff --git b/packages/utils/protobuf/codegen/cosmos/gov/v1beta1/tx.ts a/packages/utils/protobuf/codegen/cosmos/gov/v1beta1/tx.ts
index 0a7aa8cc5..2451cbfd8 100644
--- b/packages/utils/protobuf/codegen/cosmos/gov/v1beta1/tx.ts
+++ a/packages/utils/protobuf/codegen/cosmos/gov/v1beta1/tx.ts
@@ -1,7 +1,7 @@
 //@ts-nocheck
 import { Any, AnyProtoMsg, AnyAmino, AnySDKType } from "../../../google/protobuf/any";
 import { Coin, CoinAmino, CoinSDKType } from "../../base/v1beta1/coin";
-import { VoteOption, WeightedVoteOption, WeightedVoteOptionAmino, WeightedVoteOptionSDKType, TextProposal, TextProposalProtoMsg, TextProposalSDKType, voteOptionFromJSON } from "./gov";
+import { VoteOption, WeightedVoteOption, WeightedVoteOptionAmino, WeightedVoteOptionSDKType, TextProposal, TextProposalProtoMsg, TextProposalSDKType } from "./gov";
 import { MsgCreateAllianceProposal, MsgCreateAllianceProposalProtoMsg, MsgCreateAllianceProposalSDKType, MsgUpdateAllianceProposal, MsgUpdateAllianceProposalProtoMsg, MsgUpdateAllianceProposalSDKType, MsgDeleteAllianceProposal, MsgDeleteAllianceProposalProtoMsg, MsgDeleteAllianceProposalSDKType } from "../../../alliance/alliance/gov";
 import { CommunityPoolSpendProposal, CommunityPoolSpendProposalProtoMsg, CommunityPoolSpendProposalSDKType, CommunityPoolSpendProposalWithDeposit, CommunityPoolSpendProposalWithDepositProtoMsg, CommunityPoolSpendProposalWithDepositSDKType } from "../../distribution/v1beta1/distribution";
 import { ParameterChangeProposal, ParameterChangeProposalProtoMsg, ParameterChangeProposalSDKType } from "../../params/v1beta1/params";
@@ -471,7 +471,7 @@ export const MsgVote = {
       message.voter = object.voter;
     }
     if (object.option !== undefined && object.option !== null) {
-      message.option = voteOptionFromJSON(object.option);
+      message.option = object.option;
     }
     return message;
   },
diff --git b/packages/utils/protobuf/codegen/cosmwasm/wasm/v1/tx.ts a/packages/utils/protobuf/codegen/cosmwasm/wasm/v1/tx.ts
index 11d5dba92..e5ef23ea1 100644
--- b/packages/utils/protobuf/codegen/cosmwasm/wasm/v1/tx.ts
+++ a/packages/utils/protobuf/codegen/cosmwasm/wasm/v1/tx.ts
@@ -1770,7 +1770,9 @@ export const MsgInstantiateContract2 = {
       obj.funds = [];
     }
     obj.salt = message.salt ? base64FromBytes(message.salt) : undefined;
-    obj.fix_msg = message.fixMsg;
+    if (message.fixMsg) {
+      obj.fix_msg = message.fixMsg;
+    }
     return obj;
   },
   fromAminoMsg(object: MsgInstantiateContract2AminoMsg): MsgInstantiateContract2 {
diff --git b/packages/utils/protobuf/codegen/neutron/client.ts a/packages/utils/protobuf/codegen/neutron/client.ts
index 1caf8781d..9d864d753 100644
--- b/packages/utils/protobuf/codegen/neutron/client.ts
+++ a/packages/utils/protobuf/codegen/neutron/client.ts
@@ -7,21 +7,24 @@ import * as neutronFeeburnerTxRegistry from "./feeburner/tx.registry";
 import * as neutronFeerefunderTxRegistry from "./feerefunder/tx.registry";
 import * as neutronInterchainqueriesTxRegistry from "./interchainqueries/tx.registry";
 import * as neutronInterchaintxsV1TxRegistry from "./interchaintxs/v1/tx.registry";
+import * as neutronTransferV1TxRegistry from "./transfer/v1/tx.registry";
 import * as neutronCronTxAmino from "./cron/tx.amino";
 import * as neutronDexTxAmino from "./dex/tx.amino";
 import * as neutronFeeburnerTxAmino from "./feeburner/tx.amino";
 import * as neutronFeerefunderTxAmino from "./feerefunder/tx.amino";
 import * as neutronInterchainqueriesTxAmino from "./interchainqueries/tx.amino";
 import * as neutronInterchaintxsV1TxAmino from "./interchaintxs/v1/tx.amino";
+import * as neutronTransferV1TxAmino from "./transfer/v1/tx.amino";
 export const neutronAminoConverters = {
   ...neutronCronTxAmino.AminoConverter,
   ...neutronDexTxAmino.AminoConverter,
   ...neutronFeeburnerTxAmino.AminoConverter,
   ...neutronFeerefunderTxAmino.AminoConverter,
   ...neutronInterchainqueriesTxAmino.AminoConverter,
-  ...neutronInterchaintxsV1TxAmino.AminoConverter
+  ...neutronInterchaintxsV1TxAmino.AminoConverter,
+  ...neutronTransferV1TxAmino.AminoConverter,
 };
-export const neutronProtoRegistry: ReadonlyArray<[string, GeneratedType]> = [...neutronCronTxRegistry.registry, ...neutronDexTxRegistry.registry, ...neutronFeeburnerTxRegistry.registry, ...neutronFeerefunderTxRegistry.registry, ...neutronInterchainqueriesTxRegistry.registry, ...neutronInterchaintxsV1TxRegistry.registry];
+export const neutronProtoRegistry: ReadonlyArray<[string, GeneratedType]> = [...neutronCronTxRegistry.registry, ...neutronDexTxRegistry.registry, ...neutronFeeburnerTxRegistry.registry, ...neutronFeerefunderTxRegistry.registry, ...neutronInterchainqueriesTxRegistry.registry, ...neutronInterchaintxsV1TxRegistry.registry, ...neutronTransferV1TxRegistry.registry];
 export const getSigningNeutronClientOptions = ({
   defaultTypes = defaultRegistryTypes
 }: {
diff --git b/packages/utils/protobuf/codegen/regen/data/v1/tx.ts a/packages/utils/protobuf/codegen/regen/data/v1/tx.ts
index 640c2bcef..5bd95374c 100644
--- b/packages/utils/protobuf/codegen/regen/data/v1/tx.ts
+++ a/packages/utils/protobuf/codegen/regen/data/v1/tx.ts
@@ -1,5 +1,5 @@
 //@ts-nocheck
-import { ContentHash, ContentHashAmino, ContentHashSDKType, ContentHash_Graph } from "./types";
+import { ContentHash, ContentHashAmino, ContentHashSDKType, ContentHash_Graph, ContentHash_GraphAmino, ContentHash_GraphSDKType } from "./types";
 import { Timestamp } from "../../../google/protobuf/timestamp";
 import { BinaryReader, BinaryWriter } from "../../../binary";
 import { toTimestamp, fromTimestamp } from "../../../helpers";
diff --git b/packages/utils/protobuf/codegen/regen/data/v2/tx.ts a/packages/utils/protobuf/codegen/regen/data/v2/tx.ts
index fe58afb74..f0e94453b 100644
--- b/packages/utils/protobuf/codegen/regen/data/v2/tx.ts
+++ a/packages/utils/protobuf/codegen/regen/data/v2/tx.ts
@@ -1,5 +1,5 @@
 //@ts-nocheck
-import { ContentHash, ContentHashAmino, ContentHashSDKType, ContentHash_Graph } from "./types";
+import { ContentHash, ContentHashAmino, ContentHashSDKType, ContentHash_Graph, ContentHash_GraphAmino, ContentHash_GraphSDKType } from "./types";
 import { Timestamp } from "../../../google/protobuf/timestamp";
 import { BinaryReader, BinaryWriter } from "../../../binary";
 import { toTimestamp, fromTimestamp } from "../../../helpers";
